demosplan/DemosPlanCoreBundle/Command/Data/DeleteProcedureCommand.php

```diff
@@ -0,0 +1,479 @@
+            $this->processImportEmails();
+
+            // delete hashed queries
+            $this->output->writeln('Deleting hashed Queries');
+            $this->deleteHashedQueries();
+
+            // delete remaining procedure files
+            $this->output->writeln('Deleting Procedure Files');
+            $this->deleteFromTableByIdentifierArray('_files', 'procedure_id', [$this->procedureId]);
+
+            // delete procedure report entries
+            $this->output->writeln('Deleting Report Entries');
+            $this->deleteReportEntriesByIdentifierAndType('procedure', [$this->procedureId]);
+
+            // delete procedure itself
+            $this->output->writeln('Deleting Procedure');
+            $this->deleteProcedure();
+
+            // reactivate foreign key checks
+            $this->output->writeln('Activate FK Checks');
+            $this->activateForeignKeyChecks();
+
+            $this->output->writeln("Procedure $this->procedureId was purged successfully!");
+
+            return Command::SUCCESS;
+        } catch (Exception $e) {
+            $this->output->error($e->getMessage());
+            $this->output->error($e->getTraceAsString());
+
+            return Command::FAILURE;
+        }
+    }
+
+    /**
+     * Find all statements + segments
+     * Iterate over all Statements and delete related stuff like meta, attachments, etc.
+     *
+     * @throws Exception
+     */
+    private function processAllStatements(): void
+    {
+        $statementIds = array_column($this->fetchFromTableByProcedure(['_st_id'], '_statement', '_p_id'), '_st_id');
+
+        print_r($statementIds);
+
+        // delete statement meta
+        $this->output->writeln('Deleting Statement-Meta');
+        $this->deleteStatementMeta($statementIds);
+        // delete statement attachment -> files
+        $this->output->writeln('Deleting Statement-Attachments');
+        $this->processStatementAttachments($statementIds);
+        // remove all tags from statements to prepare for later tag deletion
+        $this->output->writeln('Deleting Statement-Tags');
+        $this->deleteTagsFromStatements($statementIds);
+        // delete similar statement submitter
+        $this->output->writeln('Deleting Statement-Submitters');
+        $this->deleteSimilarStatementSubmitters($statementIds);
+        // delete report entries related to statements
+        $this->output->writeln('Deleting Statement-Report-Entries');
+        $this->deleteReportEntriesByIdentifierAndType('statement', $statementIds);
+        // delete statements
+        $this->output->writeln('Deleting Statements');
+        $this->deleteFromTableByIdentifierArray('_statement', '_p_id', [$this->procedureId]);
+    }
+
+    private function processStatementAttachments(array $statementIds): void
+    {
+        $statementAttachmentQueryBuilder = $this->dbConnection->createQueryBuilder();
+        $statementAttachmentQueryBuilder
```

Comment:
Perhaps you should always check whether a table exists to make the command more robust. Scheme may change and it would be more safe regarding addons